### PR TITLE
Add `dup`

### PR DESCRIPTION
**Pull request by Russoul**
I've noticed myself using them a lot, so hopefully it makes sense to add them to the library

### CODE REVIEW

File: libs/base/Data/Maybe.idr [hunk 1]
```diff
@@ -60,3 +60,10 @@ lowerMaybe (Just x) = x
 export
 raiseToMaybe : (Monoid a, Eq a) => a -> Maybe a
 raiseToMaybe x = if x == neutral then Nothing else Just x
+
+||| Unwraps the first argument in case of `Just`
+||| returns its second argument otherwise
+export
+orElse : Maybe a -> Lazy a -> a
```
Comment (gallais) [inline libs/base/Data/Maybe.idr:67]: This is `flip fromMaybe`. I am not sure we should add such variants of existing functions. :/

File: libs/prelude/Prelude/Basics.idr [hunk 2]
```diff
@@ -25,6 +25,11 @@ id : (1 x : a) -> a           -- Hopefully linearity annotation won't
                               -- break equality proofs involving id
 id x = x
 
+||| Function that duplicates its input
+public export %inline
+dup : a -> (a, a)
```
Comment (gallais) [inline libs/prelude/Prelude/Basics.idr:30]: Given this is duplicating its argument, I am not sure we should inline it aggressively.

File: libs/base/Data/Maybe.idr (hunk 1 above)
Comment (Russoul) [reply libs/base/Data/Maybe.idr:67]: Oops, it is. I absolutely agree. What about dup ?

Comment (gallais) [reply libs/base/Data/Maybe.idr:67]: `dup` is a welcomed addition!

**Comment by Russoul** [comment]
@gallais sorry for the mess, looks like someone is going to spend the weekend learning how to use git :)

File: libs/prelude/Prelude/Basics.idr (hunk 2 above)
Comment (Russoul) [reply libs/prelude/Prelude/Basics.idr:30]: removed %inline

**Comment by gallais** [comment]
No problem.

Don't hesitate to push multiple patches to a PR even if that leads to a messy history:
when merging we can easily squash everything into a single clean commit that corresponds
to the resulting diff.

**Comment by Russoul** [comment]
Got it !

**Pull request merged**